### PR TITLE
docs: Add example for `aws_opensearchserverless` resources

### PR DESCRIPTION
**Pull request by joshjluo**
### Description

Created a comprehensive example for creating an OpenSearch Serverless collection with an encryption, network, and data access policy along with an OpenSearch Serverless-managed VPC endpoint and its required VPC resources.

### Relations

N/A

### References

- https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/opensearchserverless_access_policy
- https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/opensearchserverless_collection
- https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/opensearchserverless_security_policy
- https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/opensearchserverless_vpc_endpoint
- https://docs.aws.amazon.com/opensearch-service/latest/developerguide/serverless.html

### Output from Acceptance Testing

N/A

#### `terraform apply`

Ran `terraform apply` in the directory to confirm all resources were created

```
$ terraform apply

...

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

...

Plan: 13 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + collection_enpdoint = (known after apply)
  + dashboard_endpoint  = (known after apply)

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

...

Apply complete! Resources: 13 added, 0 changed, 0 destroyed.

Outputs:

collection_enpdoint = "..."
dashboard_endpoint = "..."
```

#### `terraform destroy`

Ran `terraform destroy` in the directory to confirm all resources were destroyed

```
$ terraform destroy

...

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  - destroy

Terraform will perform the following actions:

  ...

Plan: 0 to add, 0 to change, 13 to destroy.

Changes to Outputs:

...

Do you really want to destroy all resources?
  Terraform will destroy all your managed infrastructure, as shown above.
  There is no undo. Only 'yes' will be accepted to confirm.

  Enter a value: yes

...

Destroy complete! Resources: 13 destroyed.
```

### CODE REVIEW

**Review by johnsonaj — APPROVED** [review]
LGTM 🚀

File: examples/opensearchserverless/main.tf [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+terraform {
+  required_version = ">= 0.12"
+}
+
+# Set AWS provider region
+provider "aws" {
+  region = var.aws_region
+}
+
+# Creates an encryption security policy
```
Comment (prashanttct07) [inline examples/opensearchserverless/main.tf:14]: We don't say encryption security policy, rather we say encryption policy in our docs, Can you change it to that.

File: examples/opensearchserverless/main.tf [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+}
+
+# Creates a collection
+resource "aws_opensearchserverless_collection" "collection" {
+  name = var.collection_name
+
+  depends_on = [aws_opensearchserverless_security_policy.encryption_policy]
+}
+
+# Creates a network security policy
```
Comment (prashanttct07) [inline examples/opensearchserverless/main.tf:39]: We don't say network security policy, rather we say network policy in our docs, Can you change it to that and remove "security"

File: examples/opensearchserverless/main.tf [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+resource "aws_opensearchserverless_security_policy" "network_policy" {
+  name        = "example-network-policy"
+  type        = "network"
+  description = "public access for dashboard, VPC access for collection endpoint"
```
Comment (prashanttct07) [inline examples/opensearchserverless/main.tf:43]: dashboards -> "OpenSearch Dashboards"

**Review by prashanttct07 — COMMENTED** [review]
few minor comments, else looks good to me.

**Pull request merged**